error: member access into incomplete type 'const _GUID' with clang-cl

When building app with clang-cl compiler, compilation fails with following error:

```
2022-09-04T14:46:14.7937297Z FAILED: src/CMakeFiles/tremotesf.dir/tremotesf/filemanagerlauncher_windows.cpp.obj 
2022-09-04T14:46:14.8474303Z C:\PROGRA~1\LLVM\bin\clang-cl.exe  /nologo -TP -DFMT_UNICODE=1 -DQT_ACCESSIBILITY_SUPPORT_LIB -DQT_CORE_LIB -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050e00 -DQT_EVENTDISPATCHER_SUPPORT_LIB -DQT_FONTDATABASE_SUPPORT_LIB -DQT_GUI_LIB -DQT_MESSAGELOGCONTEXT -DQT_NETWORK_LIB -DQT_STATICPLUGIN -DQT_SVG_LIB -DQT_THEME_SUPPORT_LIB -DQT_WIDGETS_LIB -DQT_WINDOWSUIAUTOMATION_SUPPORT_LIB -DTREMOTESF_APP_ID=\"org.equeim.Tremotesf\" -DTREMOTESF_APP_NAME=\"Tremotesf\" -DTREMOTESF_BUNDLED_ICONS_DIR=\"icons\" -DTREMOTESF_BUNDLED_ICON_THEME=\"breeze\" -DTREMOTESF_BUNDLED_QT_TRANSLATIONS_DIR=\"translations\" -DTREMOTESF_QT_TRANSLATIONS_FILENAME=\"qtbase\" -DTREMOTESF_VERSION=\"1.11.3\" -DWINRT_LEAN_AND_MEAN -DWINRT_NO_MODULE_LOCK -DWINVER=0x0603 -D_SILENCE_CLANG_COROUTINE_MESSAGE -D_WIN32_WINNT=0x0603 -ID:\a\tremotesf2\tremotesf2\build-Debug-clang-cl\src -ID:\a\tremotesf2\tremotesf2\src -ID:\a\tremotesf2\tremotesf2\build-Debug-clang-cl\src\tremotesf_autogen\include -ID:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5\QtConcurrent -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5 -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5\QtCore -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\tools\qt5\mkspecs\win32-msvc -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5\QtNetwork -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5\QtWidgets -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5\QtGui -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5\QtEventDispatcherSupport -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5\QtFontDatabaseSupport -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5\QtThemeSupport -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5\QtAccessibilitySupport -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5\QtWindowsUIAutomationSupport -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\KF5\KWidgetsAddons -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\KF5 -imsvcD:\a\tremotesf2\tremotesf2\vcpkg_installed\x64-windows-static\include\qt5\QtSvg /DWIN32 /D_WINDOWS /EHsc /Zi /Ob0 /Od /RTC1 -MTd /W4 -std:c++17 /showIncludes /Fosrc\CMakeFiles\tremotesf.dir\tremotesf\filemanagerlauncher_windows.cpp.obj /Fdsrc\CMakeFiles\tremotesf.dir\ -c -- D:\a\tremotesf2\tremotesf2\src\tremotesf\filemanagerlauncher_windows.cpp
2022-09-04T14:46:14.9933983Z In file included from D:\a\tremotesf2\tremotesf2\src\tremotesf\filemanagerlauncher_windows.cpp:12:
2022-09-04T14:46:14.9935699Z In file included from C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt\winrt/Windows.Foundation.h:9:
2022-09-04T14:46:15.0114235Z ##[error]C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt\winrt/base.h(295,27): error: member access into incomplete type 'const _GUID'
2022-09-04T14:46:15.0115555Z             return { value.Data1, value.Data2, value.Data3,
2022-09-04T14:46:15.0115873Z                           ^
2022-09-04T14:46:15.0117081Z C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt\winrt/base.h(274,59): note: in instantiation of function template specialization 'winrt::guid::convert<true, _GUID>' requested here
2022-09-04T14:46:15.0117735Z         constexpr guid(GUID const& value) noexcept : guid(convert<dummy>(value)) { }
2022-09-04T14:46:15.0118123Z                                                           ^
2022-09-04T14:46:15.0119939Z C:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt\winrt/base.h(660,23): note: in instantiation of function template specialization 'winrt::guid::guid<true>' requested here
2022-09-04T14:46:15.0120512Z     inline const guid guid_v{ __uuidof(T) };
2022-09-04T14:46:15.0120798Z                       ^
```

The workaround is to include <guiddef.h> before winrt headers, but it looks like winrt/base.h should be responsible for this.

--- a/strings/base_meta.h
+++ b/strings/base_meta.h
@@ -119,12 +119,12 @@ namespace winrt::impl
 
     template <typename T>
 #if defined(__clang__)
-#if __has_declspec_attribute(uuid)
+#if __has_declspec_attribute(uuid) && defined(WINRT_IMPL_IUNKNOWN_DEFINED)
     inline const guid guid_v{ __uuidof(T) };
 #else
     inline constexpr guid guid_v{};
 #endif
-#elif defined(_MSC_VER)
+#elif defined(_MSC_VER) && defined(WINRT_IMPL_IUNKNOWN_DEFINED)
     inline constexpr guid guid_v{ __uuidof(T) };
 #else
     inline constexpr guid guid_v{};

